### PR TITLE
Some work documenting [Scene]

### PR DESCRIPTION
**Pull request by soywiz**
Some work on #80

### CODE REVIEW

File: korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt [hunk 1]
```diff
@@ -12,41 +12,106 @@ import com.soywiz.korma.geom.*
...
+/**
+ * Acts as a controller. Subclasses must override at least one of: [sceneInit] or [sceneMain].
+ *
+ * The lifecycle of the [Scene] for ([SceneContainer.pushTo], [SceneContainer.changeTo], [SceneContainer.back], [SceneContainer.forward]):
+ *
+ * - ## OLD Scene -> NEW Scene -- Scene is changed
+ * - NEW: [sceneInit] - BLOCK - Called when this scene enters. Waits until completed
+ * - NEW: [sceneMain] - DO NOT BLOCK - Called after [sceneInit]. Doesn't wait for completion. Its underlying Job is closed automatically on [sceneAfterDestroy]
+ * - OLD: [sceneBeforeLeaving] - BLOCK - Called on the old scene. Waits until completed.
+ * - Transition from the old view and the new view is performed [SceneContainer.transitionView] [View.ratio] 0..1
```
Comment (RezMike) [inline korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt]: It's unclear what the last part of the phrase means

File: korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt [hunk 2]
```diff
@@ -12,41 +12,106 @@ import com.soywiz.korma.geom.*
...
 	lateinit var sceneContainer: SceneContainer
+    /** The [ResourcesRoot] singleton. Set by the [init] method. */
 	lateinit var resourcesRoot: ResourcesRoot
-	//protected lateinit var bus: Bus
-	internal val _sceneViewContainer: Container = Container()
+
+    /** A reference to the [AG] from the [Views] */
+    val ag: AG get() = views.ag
+
+    /** A [Container] view that */
```
Comment (RezMike) [inline korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt]: Probably unfinished phrase

File: korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt [hunk 3]
```diff
@@ -12,41 +12,106 @@ import com.soywiz.korma.geom.*
...
 		_sceneViewContainer += this
 	}
 	protected open fun createSceneView(): Container = Container()
 
-	override suspend fun init(injector: AsyncInjector): Unit {
+    /**
+     * This method will be called by the [SceneContainer] that will display this [Scene].
+     * This will set the [injector], [views], [sceneContainer] and [resourcesRoot].
+     */
+    final override suspend fun init(injector: AsyncInjector) {
```
Comment (emign) [inline korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt]: init is flagged as deprecated

Comment (soywiz) [reply korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt]: Where?

Comment (emign) [reply korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt]: Here

https://github.com/korlibs/korge/blob/b3a8af5c2a2e5db8ae808629a89e970d875b6750/korge/src/commonMain/kotlin/com/soywiz/korge/scene/Module.kt#L40

Comment (soywiz) [reply korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt]: Oh I see. Not that `init`.
This `init` https://github.com/korlibs/korinject/blob/c81a6e9c04a16c4e9ad14573ea792b2d1fdcde63/korinject/src/commonMain/kotlin/com/soywiz/korinject/AsyncInjector.kt#L202

**Pull request merged**